lib/utils.coffee

```diff
@@ -293,7 +295,7 @@ exports.requestAsync = (options, retriesRemaining) ->
 	delete opts.timeout
 
 	requestTime = new Date()
-	p = fetch(url, opts)
+	p = exports.fetch(url, opts)
```

Comment:
Yes IMO it's OK and a fairly common practice to allow providing your own implementation of X if needed.